pkg/virt-controller/watch/vmi.go

```diff
@@ -977,6 +989,30 @@ func dataVolumeByNameFunc(dataVolumeInformer cache.SharedIndexInformer, dataVolu
 	}
 }
 
+func (c *VMIController) updatePVC(old, cur interface{}) {
+	curPVC := cur.(*k8sv1.PersistentVolumeClaim)
+	oldPVC := old.(*k8sv1.PersistentVolumeClaim)
+	if curPVC.ResourceVersion == oldPVC.ResourceVersion {
+		// Periodic resync will send update events for all known PVCs.
+		// Two different versions of the same PVC will always
+		// have different RVs.
+		return
+	}
+	if curPVC.DeletionTimestamp != nil {
+		return
+	}
+
+	vmis, err := c.listVMIsMatchingPVC(curPVC.Namespace, curPVC.Name)
+	if err != nil {
+		log.Log.V(4).Object(curPVC).Errorf("Error encountered during pvc update: %v", err)
+		return
+	}
+	for _, vmi := range vmis {
+		log.Log.V(4).Object(curPVC).Infof("PVC updated for vmi %s", vmi.Name)
+		c.enqueueVirtualMachine(vmi)
+	}
```

Comment:
we only actually care about enqueuing a VMI in the event that the old PVC's size doesn't match the new PVC's size, right?

Is it reasonable to only go through all this logic of matching VMIs with PVCs on every PVC update when the PVC's size changes?